spec/features/dashboard/collection_spec.rb

```diff
@@ -936,15 +939,16 @@ def get_url_fragment(type)
 
         context "to true, limits available users", js: true do
           let(:user2) { create(:user) }
-          it "to system users filted by select2" do
+          it "to system users filtered by select2" do
             visit "/dashboard/collections/#{sharable_collection_id}/edit"
             expect(page).to have_link('Sharing', href: '#sharing')
-            click_link('Sharing')
+            find_link('Sharing').click
             expect(page).to have_selector(".form-inline.add-users .select2-container")
             select_user(user2, 'Depositor')
-            click_button('Save')
-            click_link('Sharing')
-            expect(page).to have_selector('td', text: user2.user_key)
+            expect(page).to have_content "The collection's sharing options have been updated."
+            within('section.section-collection-sharing') do
+              expect(page).to have_selector('td', text: user2.user_key)
```

Comment:
Adding a user to a role saves immediately, so there is no need to Save and reclick Sharing tab.  This does check for the appropriate flash message to confirm the save happened and then confirms that the user was assigned the role.